### ISSUE TITLE
Last update seems faulty

### ISSUE THREAD

**Issue by LebedevRI** [sindresorhus/notifier-for-github#155]
Hi. With google-chrome-beta, the last update (18.7.25.917) seems kinda faulty:
1. The icon is now generic bell, instead of github-like icon.
2. Clicking on it no longer does anything. It used to open github
3. Desktop notifications no longer appear to function.

I tried removing it and re-installing, to to avail.

**Comment by mkurz** [comment]
+1
Same here. I miss the click behavior which opened https://github.com/notifications (in a new tab) before.
(I just wanted to open an issue but you were faster :wink:)

**Comment by RobertoSchneiders** [comment]
I have the same problem on Chrome 67.0.3396.99 and 68.0.3440.75

**Comment by ankurk91** [comment]
Same as #156

**Comment by ghost** [comment]
https://github.com/sindresorhus/notifier-for-github/issues/155#issuecomment-407730964
It no longer opens the notifcations page in a new tab.
All the other things I can live with.

**Comment by ravanscafi** [comment]
ping @sindresorhus

**Comment by sindresorhus** [comment]
// @notlmn @YurySolovyov

**Comment by YurySolovyov** [comment]
That moment when all tests are passing, yet your app is completely broken.

I guess the easiest way would be to revert recent changes and do a release.

**Comment by notlmn** [comment]
Sorry for all the inconvenience the last update has caused. You can find the relevant info in the commit message.

TL;DR: I was too lazy to manually test after updating dependencies!

**Comment by sindresorhus** [comment]
> The icon is now generic bell, instead of github-like icon.

That was a mistake. Fixed: https://github.com/sindresorhus/notifier-for-github/commit/67f9912a97aec1380d74497c776d2c2fb2869ed3

**Comment by mkurz** [comment]
@notlmn @sindresorhus Thanks, however when I click on the icon now, I get a popup which wants me to confirm that the extension reads my browser history. Why?

**Comment by mkurz** [comment]
Even after re-install the extension wants to get the "read browser history" permission...

**Comment by LebedevRI** [comment]
Desktop notifications still seem iffy, too. Don't show up sometimes.

**Comment by mkurz** [comment]
> Desktop notifications still seem iffy, too. Don't show up sometimes.

I can confirm that.

Why did you even update the extension? Was there a need? It was working great so far.

**Comment by notlmn** [comment]
The notifications issue is fixed in the release `18.7.27.1445`.

**Comment by mkurz** [comment]
@notlmn Thanks! Any ETA when the "read browser history" permission will be fixed? Also, clicking on the icon should open the Github notification page **in a new tab**. Thanks!

**Comment by notlmn** [comment]
Someone correct me if I'm wrong, but Chrome says "read browser history", because we request for the "tabs" permission.

Any extension that has this permission can see what tabs you have opened and track them, it is little dangerous permission. But we need that permission to open a new tab (when you click on the icon).

This is the reason, the extension doesn't ask you for the tabs permission on install or update, but only when you click on the icon. The user has to **explicitly** give the extension permission.

Chrome using that wording is almost right, but remember that the extension works as intended even you don't grant that permission.

If you still have an issues with the extension, consider open a new issue.

**Comment by YurySolovyov** [comment]
@notlmn @mkurz There is https://developer.chrome.com/extensions/permission_warnings#permissions_with_warnings

**Comment by mkurz** [comment]
Alright. But why does the extensions need the last two permissions?
https://github.com/sindresorhus/notifier-for-github/blob/ef6e7d21de0c48fa2825e07a9ec2f9294dc4440d/source/manifest.json#L16-L21
Wouldn't be `https://api.github.com/` be enough instead of `http[s]://*/*`?

**Comment by mkurz** [comment]
Those permissions
https://github.com/sindresorhus/notifier-for-github/blob/ef6e7d21de0c48fa2825e07a9ec2f9294dc4440d/source/manifest.json#L19-L20
where added just recently: https://github.com/sindresorhus/notifier-for-github/commit/4021d1fa2ca13ab24dc5f98ce45181d9e8548d35#diff-155220eb7f53e913bdaeab91a33f305b
Why?

**Comment by notlmn** [comment]
To open a tab, we don't need the tabs permission, but to update a tab by using its id filtered by URL, we need the tabs permission. We can still query all the tabs, but do not know which tab to update unless we know the URL.

This whole tab permission is mentioned in the readme already. We can remove that permission but the only reason we need is to identify and update an already opened tab instead of creating a new one each time you click on the icon.

Those additional `http[s]://*/*` permissions are redundant, we thought we need them to work on GH enterprise domains, but we don't need them, will make a PR or commit to resolve that!

**Comment by mkurz** [comment]
> To open a tab, we don't need the tabs permission, but to update a tab by using its id filtered by URL, we need the tabs permission. We can still query all the tabs, but do not know which tab to update unless we know the URL.
> 
> This whole tab permission is mentioned in the readme already. We can remove that permission but the only reason we need is to identify and update an already opened tab instead of creating a new one each time you click on the icon.
> 

I understand. However, I have a suggestion, please consider it:
Is it really that important for you to always switch to the notifications page if it is already open in an existing tab? Because of obvious privacy concerns I am not really willing to give extensions permissions to read my browser history (What if your account gets hijacked and some bad guy uploads a now malicious extension to the Chrome store? I wouldn't even recognise.) . Specifically for this case I am totally fine if clicking the icon just _always_ opens a new tab (even if in some other existing tab that page is opened already).

What do you think? Could you live with that?

**Comment by LebedevRI** [comment]
@notlmn

I *think*, clicking on the desktop notification about a new comment in an issue used to open that very issue. But now, clicking on that notification opens the https://github.com/notifications

> The notifications issue is fixed in the release 18.7.27.1445.

So i'm not sure that ^ is true..

**Comment by notlmn** [comment]
> Because of obvious privacy concerns I am not really willing to give extensions permissions to read my browser history

The [`tabs`](https://developer.chrome.com/extensions/tabs) permission doesn't let us read your browsing history (directly), that is the [`history`](https://developer.chrome.com/extensions/history) API.

An extension has access to all your tabs (excluding incognito tabs) implicitly, they can do all kinds of stuff like open new tabs, close them, and rearrange them without requiring any special permissions. But when they also have the `tabs` permission, they also get a `url` property in the tab object list provided by the browser, then that extension can track these urls and essentially _recreate_ your browsing history without needing the `history` API.

As you already know we use that permission to update and existing tab instead of opening a new one every time. I have a fairly low number of tabs open in Chrome, because you know its Chrome, but other people don't and don't know where their tabs are, instead can use the extension to jump to that tab.

I understand your suggestion to make it optional and just open a new tab every time, but that either requires us to __ask for permission on every click__ or __have an option__ in the extension options.

As this thread is already getting too long, and this discussion is not relevant here, feel free to open a new issue so everyone can add their opinion there.

**Comment by notlmn** [comment]
> > The notifications issue is fixed in the release 18.7.27.1445.
>
> So i'm not sure that ^ is true..

That release is to fix about the extension not showing any notifications at all. The issue you are facing is #159.

**Comment by mkurz** [comment]
@notlmn @sindresorhus Please have a look at pull request #162. I added a setting to alway pages in a new tab, so users with privacy concerns don't need to give the extensions the permission to read their browser history. It's opt in, the default doesn't change.
